Add more guidance on exercises/discussions/solutions (#31)
carpentries-lab/reviews: docs/reviewer_guide.md

- [ ] The lesson does not make use of superfluous data sets, e.g. increasing cognitive load for learners by introducing a new data set instead of reusing another that is already present in the lesson.
- [ ] The example tasks and narrative of the lesson are appropriate and realistic.
- [ ] The solutions to all exercises are accurate and sufficiently explained.

> - [ ] Any discussion exercises without solutions are accompanied by sufficient information/guidance that Instructors could reasonably be expected to judge whether the intended goals of the discussion have been reached.

Reviewer: I'm not sure I understand the second half of this sentence. Intended goals = learning objectives/outcomes?

```suggestion
- [ ] Any discussion exercises without solutions are accompanied by sufficient information/guidance so that Instructors can judge whether the discussion's learning objectives have been met.
```
